### PR TITLE
[FIX] account_peppol: add endpoint constraint

### PR DESCRIPTION
**Pull request by aliyatastemirova**
### [FIX] account_peppol: add endpoint constraint

The Peppol endpoint must not contain special characters. This commit adds a constraint, where the input is being sanitized. If the endpoint is not filled in, we already throw an error when
creating an edi user, so that should be fine.

task-3718573

### [FIX] account_peppol: check commercial partner instead of partner

When checking a partner's validity on Peppol network, we should always check the `commercial_partner_id` to get the right eas & endpoint. Also, the Peppol verification check should not occur if the current company does not have an active peppol registration.
Currently, if a user wants to send an invoice to a specific contact of a partner, they cannot as `_need_ubl_cii_xml` evaluates to False because the contact is not a valid Peppol participant.

opw-3748720

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account_peppol/models/res_config_settings.py [hunk 1]
```diff
@@ -53,6 +53,16 @@ class ResConfigSettings(models.TransientModel):
...
 
+    # -------------------------------------------------------------------------
+    # CONSTRAINTS
+    # -------------------------------------------------------------------------
+
+    @api.constrains('account_peppol_endpoint')
+    def _check_account_peppol_endpoint(self):
+        for company in self:
+            endpoint = company.account_peppol_endpoint or ''
+            company.account_peppol_endpoint = ''.join([char for char in endpoint if char.isalnum()])
```
Comment (smetl) [inline addons/account_peppol/models/res_config_settings.py:64]: wtf? Your constraints is 1) not checking anything and 2) is assigning a field???

Comment (aliyatastemirova) [reply addons/account_peppol/models/res_config_settings.py:64]: :sob:

File: addons/account_peppol/models/res_config_settings.py [hunk 2]
```diff
@@ -83,9 +83,15 @@ def _call_peppol_proxy(self, endpoint, params=None, edi_user=None):
         return response
 
     # -------------------------------------------------------------------------
-    # COMPUTE METHODS
+    # COMPUTE AND ONCHANGE METHODS
```
Comment (smetl) [inline addons/account_peppol/models/res_config_settings.py:86]: Put the onchange in a dedicated section plz

File: addons/account_peppol/models/res_config_settings.py [hunk 3]
```diff
@@ -83,9 +83,15 @@ def _call_peppol_proxy(self, endpoint, params=None, edi_user=None):
...
     # -------------------------------------------------------------------------
 
+    @api.onchange('account_peppol_endpoint')
+    def _onchange_account_peppol_endpoint(self):
+        for company in self:
```
Comment (smetl) [inline addons/account_peppol/models/res_config_settings.py:91]: An onchange is always for a single record. You can remove this loop

File: addons/account_peppol/models/res_config_settings.py [hunk 4]
```diff
@@ -83,9 +83,15 @@ def _call_peppol_proxy(self, endpoint, params=None, edi_user=None):
...
+            endpoint = company.account_peppol_endpoint or ''
+            company.account_peppol_endpoint = ''.join([char for char in endpoint if char.isalnum()])
```
Comment (smetl) [inline addons/account_peppol/models/res_config_settings.py:93]: `join` accepts an iterator as parameter too
```suggestion
            company.account_peppol_endpoint = ''.join(char for char in endpoint if char.isalnum())
```

**Review by smetl — APPROVED** [review]
@robodoo r+ rebase-ff

**Comment by smetl** [comment]
@robodoo r+

**Comment by aliyatastemirova** [comment]
@robodoo retry

**Pull request closed without merging**